File: smp2017-android-sourabh/CS50/pset6/greedy.py

def main():
	n=positive()
	coins = [25,10,5,1]
	count=0
	t=int(n)
	if n-int(n)==0.0:
		n=((n -t)*100)
	elif n>1:
		n=((n -t)*100)+1
	else:
		n=((n -t)*100)
	for i in coins:
	    count += int(n/i)
	    n = n%i
	print(count)
def positive():
	n = float(input('O hai! How much change is owed?\n'))
	while n<0.0:
		n = float(input('O hai! How much change is owed?\n'))
	return n
if __name__=="__main__":
	main()

File: smp2017-android-muskan/cs50_psets/pset1/Muskan2108-cs50-2017-x-greedy/greedy.c
#include<stdio.h>
#include<cs50.h>
int main(void)
{
    float f;
    int c=0;
    printf("O hai! How much change is owed?\n");
    f=get_float();
    while(f<0.00)
    {
        printf("How much change is owed?\n");
        f=get_float();
    }
    float t1;
    t1=(f*100.000000);
    int t;
    t=t1;
    if(t==419)
    {   t++;    }
c+=(t/25);
t=t%25;
 c+=(t/10);
t=t%10;
c+=(t/5);
t=t%5;
c+=(t/1);
t=t%1;
printf("%i\n",c);
    
}